core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -169,7 +169,7 @@ Long fromSnapshotId() {
     return fromSnapshotId;
   }
 
-  TableScanContext fromSnapshotId(long id) {
+  TableScanContext fromSnapshotIdExclusive(long id) {
     return new TableScanContext(snapshotId, rowFilter, ignoreResiduals,
         caseSensitive, colStats, projectedSchema, selectedColumns, options, id, toSnapshotId,
         planExecutor, fromSnapshotInclusive);
```

Comment:
nit: I think we should probably pass `false` explicitly instead of using `fromSnapshotInclusive`.

Reply:
agree. explicit `false` is more clear.